### ISSUE TITLE
Bytenode.compileCode() breaks while Bytenode.compileFile() works on same JS code...

### ISSUE THREAD

**Issue by petef19** [OsamaAbbas/bytenode#68]
Encountered a scenario here w/ an Electron main.js file where reading the JS code into a buffer and then compiling it w/ Bytenode.compileCode() created a non-working .jsc file, but using Bytenode.compileFile() on the same main.js file does work.

In either case, compiling was done inside the Electron process, where we also compiled our renderer.js file, w/ which Bytenode.compileCode() worked just fine - so it is on a case by case basis.

This is the error that was thrown when running Electron w/ the main.js file that was compiled w/ Bytenode.compileCode():

![bytenode_compile_error](https://user-images.githubusercontent.com/67035258/88273927-c70b0800-cc8f-11ea-97f3-429b75fe4b27.png)

**Comment by OsamaAbbas** [comment]
I don't know your exact setup, but you should read "README.md" document, as it explains the difference between `Bytenode.compileCode()` and `Bytenode.compileFile()`.

Either provide a minimal reproducible example, or try to turn off `compileAsModule` in `Bytenode.compileFile()`. But again, I don't know your exact situation, so I'm not sure that this will solve your issue.

**Comment by petef19** [comment]
> try to turn off `compileAsModule` in `Bytenode.compileFile()`. But again, I don't know your exact situation, so I'm not sure that this will solve your issue.

this last part of your reply confused me even further, as that is the exact opposite of what needs to be done here ;-)

but reading the docs again, it appears the main.js of our Electron app needs to be compiled as a module, hence requiring {compileAsModule: true} which is the default setting for `Bytenode.compileFile()`, which explains why `Bytenode.compileFile()` works for us w/o any special configs.

Thanks !

**Comment by OsamaAbbas** [comment]
No, I did in fact misread your first comment. I thought that `compileCode` worked and `compileFile` did not! That's why I thought turning off `compileAsModule` would either make them both work similarly or reveal the real issue.

tl;dr: never mind. Ignore my suggestion.
